akka-typed/src/main/scala/akka/typed/ActorSystem.scala

```diff
@@ -75,65 +61,93 @@ abstract class ActorSystem[-T](_name: String) extends ActorRef[T] { this: ScalaA
    */
-  implicit def executionContext: ExecutionContextExecutor = untyped.dispatcher
+  implicit def executionContext: ExecutionContextExecutor
 
   /**
    * Terminates this actor system. This will stop the guardian actor, which in turn
    * will recursively stop all its child actors, then the system guardian
    * (below which the logging actors reside).
    */
-  def terminate(): Future[Terminated] = untyped.terminate().map(t ⇒ Terminated(ActorRef(t.actor)))
+  def terminate(): Future[Terminated]
 
   /**
    * Returns a Future which will be completed after the ActorSystem has been terminated
    * and termination hooks have been executed.
    */
-  def whenTerminated: Future[Terminated] = untyped.whenTerminated.map(t ⇒ Terminated(ActorRef(t.actor)))
+  def whenTerminated: Future[Terminated]
 
   /**
    * The deadLetter address is a destination that will accept (and discard)
    * every message sent to it.
    */
-  def deadLetters[U]: ActorRef[U] = deadLetterRef
-  lazy private val deadLetterRef = ActorRef[Any](untyped.deadLetters)
+  def deadLetters[U]: ActorRef[U]
+
+  /**
+   * Create a string representation of the actor hierarchy within this system.
+   *
+   * The format of the string is subject to change, i.e. no stable “API”.
+   */
+  def printTree: String
 }
 
 object ActorSystem {
-  private class Impl[T](_name: String, _config: Config, _cl: ClassLoader, _ec: Option[ExecutionContext], _p: Props[T])
-    extends ActorSystem[T](_name) with ScalaActorRef[T] {
-    override private[akka] val untyped: ExtendedActorSystem = new ActorSystemImpl(_name, _config, _cl, _ec, Some(Props.untyped(_p))).start()
-  }
-
-  private class Wrapper(val untyped: ExtendedActorSystem) extends ActorSystem[Nothing](untyped.name) with ScalaActorRef[Nothing]
+  import internal._
 
+  /**
+   * Create an ActorSystem implementation that is optimized for running
+   * Akka Typed [[Behavior]] hierarchies—this system cannot run untyped
+   * [[akka.actor.Actor]] instances.
+   */
   def apply[T](name: String, guardianProps: Props[T],
                config:           Option[Config]           = None,
                classLoader:      Option[ClassLoader]      = None,
                executionContext: Option[ExecutionContext] = None): ActorSystem[T] = {
     val cl = classLoader.getOrElse(akka.actor.ActorSystem.findClassLoader())
     val appConfig = config.getOrElse(ConfigFactory.load(cl))
-    new Impl(name, appConfig, cl, executionContext, guardianProps)
+    new ActorSystemImpl(name, appConfig, cl, executionContext, guardianProps)
   }
 
-  def apply(untyped: akka.actor.ActorSystem): ActorSystem[Nothing] = new Wrapper(untyped.asInstanceOf[ExtendedActorSystem])
+  /**
+   * Create an ActorSystem based on the untyped [[akka.actor.ActorSystem]]
+   * which runs Akka Typed [[Behavior]] on an emulation layer. In this
+   * system typed and untyped actors can coexist.
```

Comment:
What are the current or future goal for the coexistence? What limitations? I think it's important that typed and untyped can coexists for two reasons
1) we can't convert everything in Akka to typed immediately
2) users already have something working with untyped and want to try typed for some new stuff or migrate parts of the old stuff

Feel free to move this discussion to akka-meta.